wavefront/server/packages/flo_cloud/flo_cloud/postgres/postgres.py

```
@@ -0,0 +1,261 @@
+            return
+        serialized = [
+            {
+                k: psycopg2.extras.Json(v) if isinstance(v, (dict, list)) else v
+                for k, v in row.items()
+            }
+            for row in data
+        ]
+        columns = list(serialized[0].keys())
+        col_str = ', '.join(columns)
+        val_str = ', '.join(f'%({col})s' for col in columns)
+        query = f'INSERT INTO {table_name} ({col_str}) VALUES ({val_str})'
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)